bluesky/examples.py

```diff
@@ -393,8 +411,77 @@ def trigger(self):
         return super().trigger()
 
 
+class ReaderWithFileStore(Reader):
+    """
+
+    Parameters
+    ----------
+    name : string
+    read_fields : dict
+        Mapping field names to functions that return simulated data. The
+        function will be passed no arguments.
+    conf_fields : dict, optional
+        Like `read_fields`, but providing slow-changing configuration data.
+        If `None`, the configuration will simply be an empty dict.
+    monitor_intervals : list, optional
+        iterable of numbers, specifying the spacing in time of updates from the
+        device (this applies only if the ``subscribe`` method is used)
+    loop : asyncio.EventLoop, optional
+        used for ``subscribe`` updates; uses ``asyncio.get_event_loop()`` if
+        unspecified
+    fs :FileStore
+        FileStore object that supports inserting resource and datum documents
+
+    Examples
+    --------
+    A detector that always returns 5.
+    >>> det = Readable('det', {'intensity': lambda: 5})
+
+    A detector that is coupled to a motor, such that measured insensity
+    varies with motor position.
+    >>> motor = Mover('motor')
+    >>> det = Readable('det',
+    ...                {'intensity': lambda: 2 * motor.read()['value']})
+    """
+
+    def __init__(self, *args, fs, save_path=None, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fs = fs
+        self._resource_id = None
+        if save_path is None:
+            self.save_path = mkdtemp()
+        else:
+            self.save_path = save_path
+        self.filestore_spec = 'RWFS_TIFF'  # spec name stored in resource doc
+
+        self.file_stem = new_short_uid()
+        self.path_stem = os.path.join(self.save_path, self.file_stem)
+        self.result = None
+
+    def stage(self):
+        self._resource_id = self.fs.insert_resource(self.filestore_spec,
+                                                    self.path_stem, {})
+
+    def trigger(self):
+        # save file stash file name
+        self.result = {}
+        for idx, name, val in enumerate(super().read().items()):
+            np.save('{}_{}.{}'.format(self.path_stem, idx, '.tif'), val)
```

Comment:
This names something `*..tif` You might as well hard-code the `.tif` in the format string since it's not configurable anyway.